src/main/java/doit/apihw/api/service/MemberService.java

```diff
@@ -31,26 +32,64 @@ public MemberResponse findOneMember(Long memberId) {
     // TODO : 전체 회원 정보를 조회한다.
     public List<MemberResponse> findAllMembers() {
         // DB에서 전체 회원 정보를 조회한다.
+        List<Member> members = memberRepository.findAll();
 
         // 조회된 회원 정보를 List<MemberResponse>로 변환하여 반환한다.
-        return null;
+        List<MemberResponse> memberResponses = new ArrayList<>();
+
+        for (Member member : members) {
+            MemberResponse memberResponse = MemberResponse.from(member);
+            memberResponses.add(memberResponse);
+        }
+
+        return memberResponses;
     }
 
     // TODO : 회원 이름으로 회원 정보를 조회한다.
     public List<MemberResponse> searchMembersWithName(String memberName) {
         // DB에서 memberName에 해당하는 회원 정보를 조회한다.
+        List<Member> members = memberRepository.findAllByMemberName(memberName);
 
         // 조회된 회원 정보를 List<MemberResponse>로 변환하여 반환한다.
-        return null;
+        List<MemberResponse> memberResponses = new ArrayList<>();
+        for (Member member : members) {
+            MemberResponse memberResponse = MemberResponse.from(member);
+            memberResponses.add(memberResponse);
+        }
+
+        return memberResponses;
     }
 
     // TODO : 회원 비밀번호를 변경한다.
     public void changePassword(Long memberId, AuthPasswordChangeRequest request) {
         // DB에서 memberId에 해당하는 회원 정보를 조회하고, 존재하지 않는다면 IllegalArgumentException을 발생시킨다.
+        Optional<Member> OptionalMember = memberRepository.findById(memberId);
+
+        if(OptionalMember.isEmpty()) {
+            throw new IllegalArgumentException();
+        }
+
+        Member member = OptionalMember.get();
```

Comment:
꼼꼼하게 잘 처리해주셨습니다!
Optional을 직접적으로 사용하는 것보다는 아래처럼 사용하는 것이 권장되는 편입니다.
```suggestion
Member member = memberRepository.findById(memberId)
                     .orElseThrow(() -> new IllegalArgumentException("에러 발생"));

```
[Java Optional 바르게 사용하기](https://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/
) 블로그 글 한 번 정독해보시는 것을 추천드릴게요!!